pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -180,6 +180,16 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     requestStatistics.setRequestId(requestId);
     requestStatistics.setRequestArrivalTimeMillis(System.currentTimeMillis());
 
+    // first-stage access control to prevent unauthenticated requests from using up resources
+    // secondary table-level check comes later
+    boolean hasAccess = _accessControlFactory.create().hasAccess(requesterIdentity);
+    if (!hasAccess) {
+      _brokerMetrics.addMeteredTableValue(null, BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
```

Comment:
You may want to consider a different metric here than in line 261

Reply:
imo this is the same concern. first-stage access control simply limits resource consumption (i.e. denial of service, leakage of non-table data) for unauthenticated requests. second-stage performs in-depth authorization per table. any rejections are are ACL-related.

Reply:
Ah, I know what I wanted to say:
```suggestion
      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
```

Reply:
done